core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/InfrastructureService.java

```diff
@@ -41,34 +45,46 @@
   private final NotificationEventEmitter<Infrastructure> infrastructureServiceEventEmitter;
 
   public Optional<Infrastructure> create(Infrastructure infrastructure) throws ValidationException {
-    if (infrastructureRepository.findById(infrastructure.getKey()).isPresent()) {
+    var data = convertToData(infrastructure);
```

Comment:
All the services do look very similar with all the conversion logic (as you've already mentioned), but we knew that already before. Aside from `Stream` entities (that have `Schemas`) all others are likely to be basically identical. I'm pretty sure we could have one default service impl, one for `Stream` using the default as parent with override for to handle the schema.

Reply:
I believe that is true but would prefer to dot it as a separate ticket. Have actually tried this and it involves considerable code and test changes.

Reply:
Yes can address this separately.